src/client/renderer/r_array.c

```diff
@@ -41,14 +41,26 @@ static r_array_state_t r_array_state;
 int32_t R_ArraysMask(void) {
 	uint32_t mask = R_ARRAY_MASK_VERTEX;
 
+	_Bool do_interpolation = r_view.current_entity && r_view.current_entity->old_frame != r_view.current_entity->frame && IS_MESH_MODEL(r_view.current_entity->model);
```

Comment:
To be truly safe, I would validate the frame indices against the model. Or at least `assert(ent->frame < md3->num_frames)`. etc..